### Задание №1.
Известно, что генеральная совокупность распределена нормально с известным средним квадратическим отклонением 16. Найти ширину доверительного интервала для оценки математического ожидания $\mu$ с надёжностью 0.95, если выборочное среднее равно 80, а объём выборки равен 256.

__Решение:__

Из условий задачи дано: $\sigma = 16$, $p = 0.95$, $\overline{X}=80$, $n=256$. Тогда $\alpha=1-p=0.05$.

Так как по условию нам известно, что генеральная совокупность имеет нормальное распределение, и дано среди прочего среднее квадратическое отклонение, то для вычисления ширины доверительного интервала будем использовать Z-статистику и её стандарное нормальное распределение.

$$\normalsize Z = \frac{\overline{X}-\mu}{\sigma/\sqrt{n}}$$

Границы доверительного интервала математического ожидания найдем из двойного неравенства:

$$\normalsize t_{\alpha/2} \le Z \le t_{1-\alpha/2}$$

$$\normalsize t_{\alpha/2} \le \frac{\overline{X}-\mu}{\sigma/\sqrt{n}} \le t_{1-\alpha/2}$$

$$\normalsize \frac{t_{\alpha/2}\cdot\sigma}{\sqrt{n}} \le \overline{X}-\mu \le \frac{t_{1-\alpha/2}\cdot\sigma}{\sqrt{n}}$$

$$\normalsize \overline{X}-\frac{t_{1-\alpha/2}\cdot\sigma}{\sqrt{n}} \le \mu \le \overline{X}-\frac{t_{\alpha/2}\cdot\sigma}{\sqrt{n}}$$

Тогда ширина доверительного интервала будет равна разности правой и левой границ:

$$\normalsize \Delta = \overline{X}-\frac{t_{\alpha/2}\cdot\sigma}{\sqrt{n}} - \overline{X}+\frac{t_{1-\alpha/2}\cdot\sigma}{\sqrt{n}} = 
\frac{\sigma}{\sqrt{n}}\big(t_{1-\alpha/2}-t_{\alpha/2}\big)$$

Найдем недостающие квантили порядка $\alpha/2$ и $1-\alpha/2$ через библиотеку Scipy и сразу посчитаем $\Delta$:

In [1]:
from scipy import stats

alpha = 0.05
sigma = 16
n = 256

t1 = stats.norm.ppf(alpha/2)
t2 = stats.norm.ppf(1 - alpha/2)

delta = (t2 - t1) * sigma / n ** 0.5
delta

3.9199279690801085

__Ответ:__ $\Delta\approx3.92$
___

### Задание №2.
Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет:

In [2]:
import numpy as np

sample = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что уровень значимости равен 1%? Найдите P-значение для данного теста и укажите его в качестве ответа.

__Решение:__

Для решения этой задачи будем использовать t-статистику и распределение Стьюдента:
$$\normalsize t=\frac{\overline{X}-\mu}{\sigma_X/\sqrt{n}},$$
где $\overline{X}$ - выборочное среднее, $\sigma_X$ - несмещенное выборочное среднее квадратичное отклоенение, $\mu = 200$ согласно нулевой гипотезе (утверждению продавца), $n$ - величина выборки.

Найдем значение t-статистики для нашей выборки:

In [3]:
t_stat = (sample.mean() - 200) / (sample.std(ddof=1) / np.sqrt(sample.shape[0]))
t_stat

-1.0651074037450896

Для двухсторонней критической области $\Omega_{\alpha}=(-\infty,t_{\alpha/2})\cup(t_{1-\alpha/2},+\infty)$:
$$\normalsize P = 2\cdot min\Big(F_S(S), 1-F_S(S)\Big),$$
где $F_S(S)$ - значение функции распределения при известной статистике S.

Для нашего случая значение $P$ будет равно:

In [4]:
p_value = 2 * min(stats.t.cdf(t_stat, df=sample.shape[0] - 1), 1 - stats.t.cdf(t_stat, df=sample.shape[0] - 1))
p_value

0.31456660352472854

Для проверки воспользуемся готовым t-тестом библиотеки Scipy:

In [5]:
stats.ttest_1samp(sample, popmean=200)

Ttest_1sampResult(statistic=-1.0651074037450896, pvalue=0.31456660352472854)

__Ответ:__ $P\approx 0.315$, и, сравнивая его с заданным уровнем значимости $\alpha=0.01$, можем смело принять нулевую гипотезу, то есть считать утверждение продавца о среднем весе пачки печенья в 200г верным.
___

### Задание №3.
Дана выборка диаметров подшипников:

In [6]:
samples = np.array([0.6603, 0.9466, 0.5968, 1.3792, 1.5481, 0.7515, 1.0681, 1.1134,
                    1.2088, 1.701 , 1.0282, 1.3579, 1.0191, 1.1784, 1.1168, 1.1372,
                    0.7273, 1.3958, 0.8665, 1.5112, 1.161 , 1.0232, 1.0865, 1.02  ])

Предполагая, что диаметры подшипников распределены нормально, проверьте гипотезу о том, что дисперсия случайной величины равна 0.0625 при уровне значимости $\alpha = 0.05$. Что для этого нужно знать:
1. Альтернативная гипотеза двухсторонняя.
2. Статистика для теста: $\displaystyle H = (n - 1)\frac{\sigma_X^2}{\sigma^2}$, где $n$ - число элементов в выборке, $\sigma_X^2$ - несмещённая оценка дисперсии, $\sigma^2$ - утверждаемая нулевой гипотезой дисперсия.
3. Эта статистика в предположении верности нулевой гипотезы имеет распределение хи-квадрат с параметром `df = n - 1`. Её квантили можно найти с помощью функции `scipy.stats.chi2.ppf`.

__Решение:__

Посчитаем статистику $H$ для нашей выборки: 

In [7]:
h_stat = (samples.shape[0] - 1) * samples.var(ddof=1) / 0.0625
h_stat

28.720541753333343

Посчитаем границы критической области $\Omega_\alpha$ согласно уровню значимости $\alpha = 0.05$:

In [8]:
alpha = 0.05

t1 = stats.chi2.ppf(alpha/2, df=samples.shape[0] - 1)
t2 = stats.chi2.ppf(1 - alpha/2, df=samples.shape[0] - 1)

t1, t2

(11.688551922452438, 38.0756272503558)

Критическая область $\Omega_\alpha \approx (0, 11.7)\cup(38.1, \infty)$ и посчитанная статистика $H$ для нашей выборки в эту область не попадает, следовательно нулевая гипотеза верна.

__Ответ:__ 1. Гипотеза о том, что дисперсия данной случайной величины равна 0.0625 при уровне значимости $\alpha = 0.05$, верна.